src/app/pages/admin-page/admin-page.component.html

```diff
@@ -1 +1,88 @@
+            <input matInput formControlName="city" />
+            @if (stationForm.controls.city.invalid) {
+              <mat-error>Please enter city name</mat-error>
+            }
+          </mat-form-field>
+
+          <mat-form-field subscriptSizing="dynamic" style="margin-bottom: 16px">
+            <mat-label>Latitude</mat-label>
+            <input matInput type="number" formControlName="latitude" min="-180" max="180" />
+          </mat-form-field>
+
+          <mat-form-field subscriptSizing="dynamic" style="margin-bottom: 16px">
+            <mat-label>Longitude</mat-label>
+            <input matInput type="number" formControlName="longitude" min="-180" max="180" />
+          </mat-form-field>
+
+          <div formArrayName="relations">
+            <div *ngFor="let relation of relations.controls; let i = index">
+              <mat-form-field subscriptSizing="dynamic" style="margin-bottom: 16px">
+                <mat-label>Relation {{ i + 1 }}</mat-label>
+                <mat-select [formControlName]="i">
+                  <mat-option selected>Not selected</mat-option>
+                  @if (stations$ | async; as stations) {
+                    @for (station of stations; track station.id) {
+                      <mat-option [value]="station.id" [disabled]="isOptionSelected(station.id)">{{
+                        station.city
+                      }}</mat-option>
+                    }
+                  }
+                </mat-select>
+              </mat-form-field>
+            </div>
+          </div>
+
+          <button mat-flat-button type="submit">Submit</button>
+        </form>
+      </div>
+    </div>
+    @if (stations$ | async; as stations) {
+      @for (station of visualisation$ | async; track station.id) {
+        <mat-card appearance="outlined">
+          <mat-card-header>
+            <mat-card-title>
+              <div class="title-container">
+                <span>{{ station.city }}</span>
+                <mat-icon
+                  aria-hidden="false"
+                  aria-label="delete station icon"
+                  fontIcon="delete"
+                  (click)="onDelete(station.id, stations)"
+                ></mat-icon>
+              </div>
+            </mat-card-title>
+
+            <mat-card-subtitle
+              >{{ station.latitude | number: '1.0-4' }}
+              {{ station.longitude | number: '1.0-4' }}</mat-card-subtitle
+            >
+          </mat-card-header>
+          <mat-card-content>
+            @for (connectedStation of station.connectedTo; track connectedStation.id) {
+              {{ findStationNameById(connectedStation.id, stations) }}
+            }
+          </mat-card-content>
+        </mat-card>
+      }
+      <mat-paginator
+        [pageSizeOptions]="[5, 10, 20]"
+        showFirstLastButtons
+        aria-label="Select page of periodic elements"
+      >
+      </mat-paginator>
+    }
+  </mat-card-content>
+</mat-card>
```

Comment:
Please move the stations to a separate component to insert into the Admin Panel page.